rust/lanzatool/tests/common/mod.rs

```diff
@@ -0,0 +1,148 @@
+use std::ffi::OsStr;
+use std::fs;
+use std::io::Write;
+use std::os::unix::fs::symlink;
+use std::path::{Path, PathBuf};
+use std::process::Output;
+
+use anyhow::{Context, Result};
+use assert_cmd::Command;
+use rand::distributions::Alphanumeric;
+use rand::{thread_rng, Rng};
+use serde_json::json;
+
+/// Create a mock generation link.
+///
+/// Creates the generation link using the specified version inside a mock profiles directory
+/// (mimicking /nix/var/nix/profiles). Returns the path to the generation link.
+pub fn setup_generation_link(
+    tmpdir: &Path,
+    profiles_directory: &Path,
+    version: u64,
+) -> Result<PathBuf> {
+    let toplevel = setup_toplevel(tmpdir).expect("Failed to setup toplevel");
```

Comment:
Why not propagate the error?

Reply:
There is no good reason why I didn't, just a leftover. 

But is there a good reason to propagate the error in a test? My belief was that unwrap or expect is fine there.

Reply:
Ah, yes. It's fine in the test. If you can, it's nice to propagate it anyway, because code tends to multiply and people take anything as template.

Reply:
Can you add a developer-actionable error message somehow? Right now a developer not using our shell environment will see:

```
running 1 test
test keep_only_configured_number_of_generations ... FAILED

failures:

---- keep_only_configured_number_of_generations stdout ----
thread 'keep_only_configured_number_of_generations' panicked at 'Failed to setup generation link: Failed to setup toplevel

Caused by:
    environment variable not found', tests/gc.rs:18:18
```

Alternatively a well placed comment in the code would also be ok.